control/margins.py

```diff
@@ -300,6 +321,27 @@ def stability_margins(sysdata, returnall=False, epsw=0.0):
         raise ControlMIMONotImplemented(
             "Can only do margins for SISO system")
 
+    if method == 'frd':
+        # convert to FRD if we got a transfer function
+        if isinstance(sys, xferfcn.TransferFunction):
+            omega_sys = freqplot._default_frequency_range(sys)
+            if sys.isctime():
+                sys = frdata.FRD(sys, omega_sys)
+            else:
+                omega_sys = omega_sys[omega_sys < np.pi / sys.dt]
+                sys = frdata.FRD(sys(np.exp(1j*sys.dt*omega_sys)), omega_sys,
```

Comment:
Are you trying to hide the fact, that `FRD(sys, omega)` does not handle discrete time systems correctly? ;)

Reply:
See #568.